Add "Running geomedian on grouped or resampled timeseries" section

andrewdhicks opened the pull request:
### Proposed changes
Added a section on applying the geomedian function using `resample`/`groupby`.

Seeking feedback on:

1. Is the `Dataset -> reshape_for_geomedian -> resample -> reduce -> to_dataset` process too complex? Should we wrap it up into a function?

2. The other composite notebook doesn't cover `groupby`. Should we remove reference to it here, or add it to the composite notebook?

3. Does the text make sense?

### Checklist (replace `[ ]` with `[x]` to check off)
- [X] Remove any unused Python packages from `Load packages`
- [X] Remove any unused/empty code cells
- [X] Remove any guidance cells (e.g. `General advice`)
- [X] Ensure that all code cells follow the [PEP8 standard](https://www.python.org/dev/peps/pep-0008/) for code. The `jupyterlab_code_formatter` tool can be used to format code cells to a consistent style: select each code cell, then click `Edit` and then one of the `Apply X Formatter` options (`YAPF` or `Black` are recommended).
- [ ] Include relevant tags in the final notebook cell and re-use tags if possible
- [X] Clear all outputs, run notebook from start to finish, and save the notebook in the state where all cells have been sequentially evaluated

robbibt commented:
Hey @andrewdhicks , this is a great addition. Some thoughts:

1. The new geomedian code does bring lots of pre- and post-processing functions with it (the rescaling/`to_f32` code, `from_float`, and now `reshape_for_geomedian`, `.reduce()` and `.to_dataset()`). I'm actually OK with the `resample-reduce-to_dataset` bit (as it's all standard `xarray` API), but it's the `to_f32`, `from_float` and `reshape_for_geomedian` funcs that I think have the potential to be the most confusing. Is there any overlap between these funcs that we could simplify (e.g. replace the `to_f32` bits with `reshape_for_geomedian`)? If not, we might want to add in a bit more basic explanation of what the functions do and why they are important, e.g. after `da_scaled = reshape_for_geomedian(ds_scaled)`, actually print out `da_scaled` and explain what the function did.

2. `groupby` does seem like a good fit for the other notebook, as it's a fundamental `xarray` technique that I think would be best introduced in a basic notebook... this one is a bit more complex and is unlikely to be the place users will want to first find out about `groupby`.

3. Some typos:
- `(represented by the 'S' at the end)` No full stop
- `using the dask dahboard.` Spelling for dashboard
- `and see the chnage in the landscape over the year:` Spelling for change

4. More generally, this notebook can probably be updated to load less data... 3 minutes for the geomedian calculation is long for a demo notebook.

andrewdhicks commented:
Thanks for the feedback @robbibt !

I've wrapped the to/from float and `xr_geomedian` functions into a helper function that can work with `map` on resample/groupby objects.  It doesn't expose all of the parameters of `xr_geomedian`, but we can add these if needed.

I've also added a section on `groupby` to the more general "Generating Composites" notebook.
Typos also fixed. I've reduced the area a bit, but not sure if it is fast enough. We may need to pick a different location if we want to see meaningful changes.

andrewdhicks commented:
Thanks @cbur24. 

The issue was that the returned per-pixel `time` co-ordinates for each band were different and couldn't be merged, as there were some pixels that didn't have data in every band. This usually gets masked out in Landsat with the PQ continuity bit, but I don't know if S2 has an equivalent.

I've changed it so does explicit masking across every band first and it working great now.
